Latest espeak to resolve italian pronunciation

feerrenrut opened the pull request:
Update Espeak to commit 5226ff0

Some changes had to be made to build system so support this Espeak update.
We must now compile and link and include in the search paths `ucd-tools`. To do this, I diffed the the visual studio project files in the espeak project.

I have excluded 'zhy` dictionary since it was failing to build. 

A try build has been given to the user asking for better support, we have not yet had feedback.

feerrenrut committed:
Update espeak-ng

This is to see if it resolves issues with pronunciation in Italian

feerrenrut committed:
Get espeak building again

zhy has been excluded as it wont compile.

jcsteh commented:
This update introduces some ugly regressions. According to espeak-ng/espeak-ng#250, at least some of these should be fixed in the latest espeak-ng commit. However, @derekriemer noted on Gitter that he can't get this commit to build with NVDA:

> ```
> ctype.obj : error LNK2019: unresolved external symbol _ucd_properties referenced
> in function _ucd_isalnum
> readclause.obj : error LNK2019: unresolved external symbol _clause_type_from_cod
> epoint referenced in function _ReadClause
> build\x86\espeak\espeak.dll : fatal error LNK1120: 2 unresolved externals
> ```

@derekriemer also did a bisect and discovered that this occurs in the very first espeak-ng commit after the one we now use in next.

@feerrenrut, can you take a look at updating this PR to the latest espeak-ng commit? Thanks.

feerrenrut committed:
Update to the latest espeak

Update to commit eb3c9a9bbd186ec3ead40059fbe865eed1cd3772 of Espeak-ng

See
https://github.com/espeak-ng/espeak-ng/commits/eb3c9a9bbd186ec3ead40059fbe865eed1cd3772

feerrenrut committed:
Fix build errors with latest espeak.

Brian1Gaff commented:
This may well be totally unrelated to this as its an English issue, but.
 When  the
% symbol is spoken it obeys the symbol  pronunciation in the list if its 
actually the normal
% in a text string or static on a page, However when spoken as a progress 
bar, it changes slightly. Though not a big issue it seems that the way this 
is spoken has recently changed in Espeak and thus you hear the two versions 
one affected by the symbols list and one not.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

feerrenrut committed:
Update espeak to fix #7167

feerrenrut committed:
Allow unicode characters in espeak language names

Fixes part of #7167
selecting Norwegian Bokmål no longer causes an error.

jcsteh commented on source/synthDrivers/espeak.py:
```diff
@@ -193,9 +193,11 @@ def _getAvailableVoices(self):
 		voices=OrderedDict()
 		for v in _espeak.getVoiceList():
 			l=v.languages[1:]
+			# #7167: Some languages names require unicode characters EG: Norwegian Bokmål
```
nit: I think this would be more readable as "Some language names contain Unicode characters".

jcsteh reviewed (approved):
Looks good!

feerrenrut committed:
Review actions for #7115

Clarified comment

feerrenrut committed:
Update espeak-ng

To commit c5afdbe0d1292a6f1d95e3794b62e4be93d7a6cd

feerrenrut commented:
The changes for this PR are going to be reverted from 'next'. We will first incubate the most recent release of espeak-ng [1.49.1](https://github.com/espeak-ng/espeak-ng/releases/tag/1.49.1). Once that is incubated and (if considered stable) merged to master, we will resume updating next on a regular basis. For now I will close this PR.
